Tools/CMake/AMReXParallelBackends.cmake

```
@@ -193,21 +194,40 @@ if (AMReX_HIP)
 
    set(CMAKE_MODULE_PATH "${HIP_PATH}/cmake" ${CMAKE_MODULE_PATH})
 
-   find_package(HIP REQUIRED)
+   find_package(hip)
 
    if("${HIP_COMPILER}" STREQUAL "hcc")
       message(FATAL_ERROR "Using (deprecated) HCC compiler: please update ROCm")
    endif()
 
-   if(HIP_FOUND)
+   if(hip_FOUND)
       message(STATUS "Found HIP: ${HIP_VERSION}")
-      message(STATUS "HIP: Platform=${HIP_PLATFORM} Compiler=${HIP_COMPILER} Path=${HIP_PATH}")
+      message(STATUS "HIP: Runtime=${HIP_RUNTIME} Compiler=${HIP_COMPILER} Path=${HIP_PATH}")
    else()
       message(FATAL_ERROR "Could not find HIP."
          " Ensure that HIP is either installed in /opt/rocm/hip or the variable HIP_PATH is set to point to the right location.")
    endif()
 
-   # avoid forcing the rocm LLVM flags on a gfortran
+   if(${_this_comp} STREQUAL hipcc AND NOT AMReX_FORTRAN)
+       message(WARNING "You are using the legacy wrapper 'hipcc' as the HIP compiler.\n"
+           "This is only needed when building with Fortran support and with ROCm/HIP <=4.2.0. "
+           "Use AMD's 'clang++' compiler instead.")
+   endif()
+   if(NOT ${_this_comp} STREQUAL hipcc)
+       target_link_libraries(amrex PUBLIC hip::device)
+
+       # work-around for https://github.com/ROCm-Developer-Tools/HIP/issues/2278
+       # CXX_STANDARD always adds -std=c++XX, even if the compiler default fulfills it
+       set_property(TARGET amrex PROPERTY CXX_STANDARD 14)
+
+       # work-around for ROCm <=4.2
+       # https://github.com/ROCm-Developer-Tools/HIP/pull/2190
+       target_compile_options(amrex PUBLIC
+          "$<$<COMPILE_LANGUAGE:CXX>:SHELL:-mllvm;-amdgpu-early-inline-all=true;-mllvm;-amdgpu-function-calls=false;-x hip>"
```

See comment in https://github.com/ROCm-Developer-Tools/HIP/pull/2280/files#r641120476
```suggestion
          "$<$<COMPILE_LANGUAGE:CXX>:SHELL:-mllvm;-amdgpu-early-inline-all=true;-mllvm;-amdgpu-function-calls=false;-xhip>"
```